File: c++/1018.cpp
#include <iostream>
#include <string>

using namespace std;

int main(){
  int n, m, count, minValue = 32;
  char board[50][50] = {};
  cin >> n >> m;

  for(int i=0; i<n; i++){
    string s;
    cin >> s;
    for(int j=0; j<m; j++){
      board[i][j] = s[j];
    }
  }

  for(int i=0; i<n-7; i++){
    for(int j=0; j<m-7; j++){
      count = 0;
      for(int a=i; a<i+8; a++){
        for(int b=j; b<j+8; b++){
          if(((a+b)%2 == 0 && board[a][b] == 'B') || (a+b)%2 != 0 && board[a][b] == 'W'){
            count += 1;
          } 
        }
      }
      if(minValue > min(count, 64-count)){
        minValue = min(count, 64-count);
      }
    }
  }
  cout << minValue;
  
  return 0;
}

File: js/1018.js
const fs = require('fs');

const inputs = fs.readFileSync('/dev/stdin').toString().split('\n');
const [m, n] = inputs[0].split(' ').map(Number);

const board = [];
let result = m * n;

for (let i = 1; i <= m; i++) {
  board.push(inputs[i]);
}

for (let i = 0; i <= m - 8; i++) {
  for (j = 0; j <= n - 8; j++) {
    const a = checkBoard(i, j);
    result = Math.min(result, a, 64 - a);
  }
}

function checkBoard(y, x) {
  let cnt = 0;
  for (let i = y; i < y + 8; i++) {
    for (let j = x; j < x + 8; j++) {
      if (
        ((i + j) % 2 === 0 && board[i][j] === 'B') ||
        ((i + j) % 2 === 1 && board[i][j] === 'W')
      )
        cnt++;
    }
  }
  return cnt;
}

console.log(result);
